### ISSUE TITLE
Multiple r:deps() in a single query should be ANDed, not ORed

### ISSUE THREAD

**Issue by tsenart** [sourcegraph/sourcegraph#32998]

**Comment by sourcegraph-bot-2** [comment]
Heads up @jjeffwarner - the "team/search-core" label was applied to this issue.

**Comment by mrnugget** [comment]
Just to add more context to this ticket, after I looked into it with @stefanhengl.

Say we have the following dependency graph:

```
 pkg A ──┬┬─▶ pkg B ──┬─▶ pkg E 
         ││           │         
         ││           │         
         └┼─▶ pkg C ──┘         
          │                     
          │                     
          └─▶ pkg D             
```

The problem this ticket describes is that if I query

```
repo:deps(pkgA) repo:deps(pkgB)
```

I get 

```
pkgB
pkgC
pkgD
pkgE
```

which is the union of the dependencies. What the I want though is the intersection: what are the dependencies of `pkgA` _and_ dependencies of `pkgB`?

---

Looking at the code we also realised that once control reaches this point here it's already too late:

https://github.com/sourcegraph/sourcegraph/blob/b61aa924da728db229e3b41d97b014a5c9efc791/internal/search/repos/repos.go#L128

because `op.Dependencies` will contain `pkgA` and `pkgB` and then just concatenate the results.
